src/nationalarchives/components/nested-navigation/nested-navigation.ts

```diff
@@ -163,25 +163,30 @@ export class NestedNavigation {
       localStorage.setItem(`${inputType}-state`, JSON.stringify({ expanded }));
   };
 
-  setSelected: (li: HTMLLIElement | null, inputType: InputType) => void = (
+  setSelected: (li: HTMLLIElement, inputType: InputType) => void = (
     li,
     inputType
   ) => {
     if (
       li == null ||
-      (li.id.includes("folder") && inputType == InputType.radios)
+      (li.querySelectorAll("ul").length > 0 && inputType == InputType.radios)
     ) {
+      this.toggleNode(li as HTMLLIElement, li.id, InputType.radios);
```

Comment:
You can get here if `li` is null but if it is null, then the call to `li.id` will fail with an error. It looks like `li` isn't ever null so maybe you can just remove the `li == null` check.

Reply:
Good spot. I removed and tidied up/commented this function a bit. It's got a bit convoluted since we added radios + checkboxes. Tried to sense check and removed another redundant conditional branch. Would you mind checking this again please @MancunianSam ?